src/pokemon.c

```diff
@@ -6424,3 +6424,95 @@ u16 GetSpeciesPreEvolution(u16 species)
 
     return SPECIES_NONE;
 }
+
+u32 CreateCustomMon(u16 species, u8 level, u16 item, u8 ball, u8 nature, u8 abilityNum, u8 gender, u8 *evs, u8 *ivs, u16 *moves, bool8 isShiny)
+{
+    u16 nationalDexNum;
+    int sentToPc;
+    struct Pokemon mon;
+    u32 i;
+
+    // check whether to use a specific nature or a random one
+    if (nature >= NUM_NATURES)
+        nature = Random() % NUM_NATURES;
+
+    // create a Pokémon with basic data
+    if (gender == MON_MALE || gender == MON_FEMALE)
+        CreateMonWithGenderNatureLetter(&mon, species, level, 32, gender, nature, 0);
+    else
+        CreateMonWithNature(&mon, species, level, 32, nature);
+
+    if (isShiny)
+        SetMonData(&mon, MON_DATA_IS_SHINY, &isShiny);
+
+    for (i = 0; i < NUM_STATS; i++)
+    {
+        // EV
+        if (evs[i] <= MAX_PER_STAT_EVS)
+            SetMonData(&mon, MON_DATA_HP_EV + i, &evs[i]);
+
+        // IV
+        if (ivs[i] <= MAX_PER_STAT_IVS)
+            SetMonData(&mon, MON_DATA_HP_IV + i, &ivs[i]);
```

Comment:
When you set an IV value to over 31 it will assign a random value. I think it would be better if defaulted to 31 if something invalid was given.

Reply:
It was indeed intended. Imo there *should* be a way to generate a random value, so I figured that anything beyond 31 would be acceptable for that.

If you have a better idea on to handle that lmk though :eyes:

Reply:
No, that's fine.